### PR TITLE
Feature: add proper boot partition config files support

### PR DESCRIPTION
**Pull request by ccrisan**

### CODE REVIEW

**Comment by ccrisan** [comment]
@jasaw based on your recent work on `wpa_supplicant.conf` boot support, I have generalized it and made it available for all (most of the) configuration files. If you find some spare time, could you please take a look over this PR and let me know if you find anything that you would fix/change or do otherwise?

File: board/common/overlay/etc/init.d/base [hunk 1]
```diff
@@ -24,3 +24,37 @@ msg_background() {
...
+        test $RO == 0 && mount -o remount,rw /boot
+        rm -rf $user_conf
+        test $RO == 0 && mount -o remount,ro /boot
+    fi
+
+    if [[ ! -e "$actual_conf" && -e "$system_conf" ]]; then
+        cp "$system_conf" "$actual_conf"
+    fi
+
+    # we want only Unix newlines in conf files
```
Comment (jasaw) [inline board/common/overlay/etc/init.d/base]: @ccrisan I would move this part into the above `if [[ -n "$user_conf" && -e "$user_conf" ]]; then` block, just after `cp -rf "$user_conf" "$actual_conf"`, to minimize the amount of processing required on every start up. We only need to do the newline conversion if the file coming from the user. Our system default config files should already have unix newline.

**Comment by jasaw** [comment]
I like the idea of generalizing the user config file handling and re-using it for most config files. Brilliant.
This PR looks good from code inspection, with one comment regarding the newline conversion.

File: board/common/overlay/etc/init.d/base (hunk 1 above)
Comment (ccrisan) [reply board/common/overlay/etc/init.d/base]: Very good point. Fixed via 92bd4b031d8ac63928939290c852df64311fc518.

**Comment by ccrisan** [comment]
All the "global" variables that act as constants have just been uppercased, like one would expect them to be.

I'm planning on replacing all `if [...]` statements with `if [[...]]` ones to prevent the extra process invocation. The same applies to `test`. I'm hoping to see a slightly faster boot process.

**Pull request merged**